### DISCUSSION TITLE
Best way to (re)implement file-loader?name query in Asset Modules

### DISCUSSION THREAD

**Discussion by nkalinov** [webpack/webpack#15472] [General]
For certain assets in our project we don't want the exported filename to change so we used to override the default file-loader asset name setting via an inline loader syntax: `import File from '-!file-loader?name=[name].[ext]!./file.ext';`

I would like to reimplement the same feature in Asset Modules and here is what I have so far using `generator.filename`. 
```
const queryString = require('query-string');

// ... module.rules
{
  type: 'asset/resource',
  generator: {
  filename: (pathData, assetInfo) => {
    const query = pathData.module.resourceResolveData.query;
    const { name } = queryString.parse(query.slice(1));
    return name || 'static/media/[name].[hash][ext]';
  },
},
```
And it works: `import File from './file.ext?name=[name][ext]';`

My question is if there is a better way to do that?

**Comment by alexander-akait** [comment]
That is valid solution
